### PR TITLE
Import Datadog modules

### PR DESCRIPTION
**Pull request by xp-1000**
This PR aims to import equivalent terraform modules from Datadog https://github.com/claranet/terraform-datadog-integrations

It will also try to improve some modules providing more customizations and make it possible to deploy only signalfx part for cloud provider modules.

### CODE REVIEW

File: alerting/pagerduty/integrations-pagerduty.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resource "signalfx_pagerduty_integration" "sfx_integration" {
+  name = "PagerdutyIntegration${var.suffix == "" ? "" : "-${title(var.suffix)}"}"
+	enabled = var.enabled
```
Comment (Shr3ps) [inline alerting/pagerduty/integrations-pagerduty.tf]: tf fmt?

Comment (xp-1000) [reply alerting/pagerduty/integrations-pagerduty.tf]: already done but not pushed sorry, now everything is ready for review

File: cloud/gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# CLOUD GCP DataDog integrations
+
+## How to use this module
+
+```hcl
+module "datadog-integrations-cloud-gcp" {
```
Comment (pdecat) [inline cloud/gcp/README.md]:
```suggestion
module "signalfx-integrations-cloud-gcp" {
```

File: cloud/gcp/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# CLOUD GCP DataDog integrations
```
Comment (pdecat) [inline cloud/gcp/README.md]:
```suggestion
# CLOUD GCP SignalFx integrations
```

File: alerting/pagerduty/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+output "sfx_integration_id" {
+  description = "SignalFX integration ID"
```
Comment (pdecat) [inline alerting/pagerduty/outputs.tf]:
```suggestion
  description = "SignalFx integration ID"
```

File: cloud/aws/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
+# CLOUD AWS SignalFX integrations
```
Comment (pdecat) [inline cloud/aws/README.md]:
```suggestion
# CLOUD AWS SignalFx integrations
```

File: cloud/aws/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  value       = aws_iam_role.sfx_role.arn
+}
+
+output "aws_role_name" {
+  description = "The IAM role name of the SignalFx integration"
+  value       = aws_iam_role.sfx_role.name
+}
+
+output "sfx_integration_id" {
+  description = "SignalFX integration ID"
```
Comment (pdecat) [inline cloud/aws/outputs.tf]:
```suggestion
  description = "SignalFx integration ID"
```

File: cloud/aws/outputs.tf [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  value       = signalfx_aws_integration.sfx_integration.integration_id
+}
+
+output "sfx_external_id" {
+  description = "SignalFX integration external ID"
```
Comment (pdecat) [inline cloud/aws/outputs.tf]:
```suggestion
  description = "SignalFx integration external ID"
```

File: cloud/gcp/README.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+```
+
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:-----:|
+| enabled | Whether the GCP integration is enabled | `string` | `"true"` | no |
+| gcp\_project\_id | GCP project id for use with the SignalFX GCP integration | `string` | n/a | yes |
+| gcp\_service\_account\_id | GCP service account id for use with the SignalFX GCP integration | `string` | `""` | no |
```
Comment (pdecat) [inline cloud/gcp/README.md]:
```suggestion
| gcp\_project\_id | GCP project id for use with the SignalFx GCP integration | `string` | n/a | yes |
| gcp\_service\_account\_id | GCP service account id for use with the SignalFx GCP integration | `string` | `""` | no |
```

File: cloud/gcp/permissions.tf [hunk 9]
```diff
@@ -0,0 +1,22 @@
+resource "google_project_iam_custom_role" "sfx_role" {
+  role_id     = "signalfx${var.suffix == "" ? "" : ".${substr(lower(replace(var.suffix, "-", "_")), 0, 64)}"}"
+  title       = "SignalFX${var.suffix == "" ? "" : " - ${title(var.suffix)}"}"
+  description = "SignalFX viewer role for monitoring${var.suffix == "" ? "" : " for ${lower(var.suffix)}"}"
```
Comment (pdecat) [inline cloud/gcp/permissions.tf]:
```suggestion
  description = "SignalFx viewer role for monitoring${var.suffix == "" ? "" : " for ${lower(var.suffix)}"}"
```

File: cloud/gcp/service-accounts.tf [hunk 10]
```diff
@@ -0,0 +1,19 @@
+data "google_service_account" "sfx_service_account" {
+  count      = var.gcp_service_account_id != "" ? 1 : 0
+  account_id = var.gcp_service_account_id
+}
+
+resource "google_service_account" "sfx_service_account" {
+  count        = var.gcp_service_account_id == "" ? 1 : 0
+  account_id   = "signalfx${var.suffix == "" ? "" : "-${substr(lower(var.suffix), 0, 30)}"}"
+  display_name = "SignalFX Integration${var.suffix == "" ? "" : " - ${title(var.suffix)}"}"
```
Comment (pdecat) [inline cloud/gcp/service-accounts.tf]:
```suggestion
  display_name = "SignalFx Integration${var.suffix == "" ? "" : " - ${title(var.suffix)}"}"
```

File: cloud/gcp/variables.tf [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  default     = "true"
+}
+
+variable "poll_rate" {
+  description = "GCP poll rate in seconds (One of 60 or 300)"
+  default     = 300
+}
+
+variable "gcp_service_account_id" {
+  description = "GCP service account id for use with the SignalFX GCP integration"
```
Comment (pdecat) [inline cloud/gcp/variables.tf]:
```suggestion
  description = "GCP service account id for use with the SignalFx GCP integration"
```

File: cloud/gcp/variables.tf [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  type        = string
+  default     = ""
+}
+
+variable "gcp_project_id" {
+  description = "GCP project id for use with the SignalFX GCP integration"
```
Comment (pdecat) [inline cloud/gcp/variables.tf]:
```suggestion
  description = "GCP project id for use with the SignalFx GCP integration"
```

File: database/mysql/README.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
+# DATABASE MYSQL SignalFX integrations
```
Comment (pdecat) [inline database/mysql/README.md]:
```suggestion
# DATABASE MYSQL SignalFx integrations
```

File: database/mysql/README.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+| monitoring\_user\_password | Monitoring user password if created by the module |
+
+## Related documentation
+
+[Official documentation](https://docs.signalfx.com/en/latest/integrations/integrations-reference/integrations.mysql.html)
+[Collectd documentation](https://collectd.org/wiki/index.php/Plugin:MySQL)
+
+## Purpose
+
+This module prepares a MySQL server for use with SignalFX.
```
Comment (pdecat) [inline database/mysql/README.md]:
```suggestion
This module prepares a MySQL server for use with SignalFx.
```

File: cloud/aws/roles.tf [hunk 15]
```diff
@@ -0,0 +1,10 @@
+resource "aws_iam_role" "sfx_role" {
+  name               = "SignalFxIntegration${var.suffix == "" ? "" : "-${title(var.suffix)}"}"
+  description        = "signalfx integration to read out data and send it to signalfxs aws account"
```
Comment (pdecat) [inline cloud/aws/roles.tf]:
```suggestion
  description        = "SignalFx integration to read out data and send it to SignalFx's AWS account"
```

File: cloud/aws/README.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+| sfx\_external\_id | SignalFx integration external ID |
+| sfx\_integration\_id | SignalFx integration ID |
+
+## Related documentation
+
+[Official documentation](https://docs.signalfx.com/en/latest/integrations/amazon-web-services.html#connect-to-aws-cloudwatch)
+
+## Requirements
+
+You need to configure you AWS provider.
```
Comment (rossifumax) [inline cloud/aws/README.md]:
```suggestion
You need to configure your AWS provider.
```

**Pull request merged**